core/types.rs

```diff
@@ -873,6 +1103,18 @@ pub fn compare_immutable_to_record(
     Ordering::Equal
 }
 
+pub fn compare_immutable(l: &[RefValue], r: &[RefValue]) -> std::cmp::Ordering {
```

Comment:
Can the implementation of this function just be `l.partial_cmp(r).unwrap()`? What's the difference? Also, are the `impl PartialOrd<OwnedValue> for RefValue` implementations also unnecessary now?